### PR TITLE
MODKBEKBJ-21 - Implement GET /eholdings/providers/{provider_id}/packages

### PR DESCRIPTION
**Pull request by ivan-brovenko**
## Purpose
Per https://issues.folio.org/browse/MODKBEKBJ-21 we want to have the ability to get packages by provider id by supporting  `/eholdings/providers/{provider_id}/packages` endpoint

## Approach

- Implement get provider packages method
- Added permissions to the request
- Mapped packages returned from RM API by provider id to a collection of packages

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -50,7 +50,8 @@
         },
         {
           "methods": ["GET"],
-          "pathPattern": "/eholdings/providers/{providerId}/packages"
+          "pathPattern": "/eholdings/providers/{providerId}/packages",
+          "permissionRequired": ["kb-ebsco.provider.packages.get"]
```
Comment (andrii-paias) [inline descriptors/ModuleDescriptor-template.json]: There's a typo, it should be "permissionsRequired"

File: src/main/java/org/folio/rest/impl/EholdingsProvidersImpl.java [hunk 2]
```diff
@@ -180,8 +191,45 @@ public void putEholdingsProvidersByProviderId(String providerId, String contentT
...
+  @Validate
   @HandleValidationErrors
-  public void getEholdingsProvidersPackages(String q, String filterSelected, String filterType, String sort, int page, int count, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(Future.succeededFuture(GetEholdingsProvidersResponse.status(Response.Status.NOT_IMPLEMENTED).build()));
+  public void getEholdingsProvidersPackagesByProviderId(String providerId, String q, String filterSelected,
+                                                        String filterType, String sort, int page, int count,
+                                                        Map<String, String> okapiHeaders,
+                                                        Handler<AsyncResult<Response>> asyncResultHandler,
+                                                        Context vertxContext) {
+    long providerIdLong;
```
Comment (andrii-paias) [inline src/main/java/org/folio/rest/impl/EholdingsProvidersImpl.java]: In all other cases we validate that q is not empty and return an error if it's empty, I think it's better to do it here too, for consistency

File: ramls/examples/providers/providers_providerId_packages_get_400_response.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "errors": [
+    {
+      "title": "Parameter OrderBy should be by PackageName when parameter Search is empty."
```
Comment (andrii-paias) [inline ramls/examples/providers/providers_providerId_packages_get_400_response.json]: error message "Parameter OrderBy should be by PackageName when parameter Search is empty" is an error returned from RM API, mod-kb-ebsco-java doesn't have parameter OrderBy, in other endpoints we return "Search parameter cannot be empty" for this case

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -197,6 +203,7 @@
         "kb-ebsco.providerCollection.get",
         "kb-ebsco.provider.get",
         "kb-ebsco.provider.put",
+        "kb-ebsco.provider.packages.get",
```
Comment (carolegodfrey) [inline descriptors/ModuleDescriptor-template.json]: This is very minor -- but just wonder if we should be calling this permission
`kb-ebsco.provider.packageCollection.get`?  or if `kb-ebsco.provider.packages.get` is more readable

File: ramls/providers.raml [hunk 5]
```diff
@@ -88,20 +89,36 @@ traits:
...
+    /packages:
+        get:
+          description: Search within a list of packages associated with a given provider.
+          is: [queriable, filterable, sortable, pageable]
+          responses:
+            200:
+              description: OK
+              body:
+                application/vnd.api+json:
+                  description: List of providers matching the provider name and the total number of providers found.
```
Comment (carolegodfrey) [inline ramls/providers.raml]: Minor adjustment - Should this description be something more like
`List of packages associated with a given provider matching search criteria`

Comment (ivan-brovenko) [reply ramls/providers.raml]: Thanks, updated

File: descriptors/ModuleDescriptor-template.json (hunk 4 above)
Comment (ivan-brovenko) [reply descriptors/ModuleDescriptor-template.json]: Thanks, updated

File: ramls/examples/providers/providers_providerId_packages_get_400_response.json (hunk 3 above)
Comment (ivan-brovenko) [reply ramls/examples/providers/providers_providerId_packages_get_400_response.json]: Thanks, updated

File: src/main/java/org/folio/rest/impl/EholdingsProvidersImpl.java (hunk 2 above)
Comment (ivan-brovenko) [reply src/main/java/org/folio/rest/impl/EholdingsProvidersImpl.java]: Thanks, updated

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (ivan-brovenko) [reply descriptors/ModuleDescriptor-template.json]: Thanks, updated

**Pull request merged**